# HG changeset patch
# User Mark Lodato <lodatom@gmail.com>
# Date 1255321067 14400
freeze: Makefile: use -fno-strict-aliasing

Python 2 cannot compile with strict aliasing, so the Cython modules
should really be compiled with -fno-strict-aliasing.  See Issue #395.

diff --git a/Demos/freeze/Makefile b/Demos/freeze/Makefile
index fe4f36e..de593da 100644
--- a/Demos/freeze/Makefile
+++ b/Demos/freeze/Makefile
@@ -8,7 +8,7 @@ PY_LDFLAGS = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_va
 PY_CPPFLAGS = $(shell $(PYTHON) -c 'from distutils.sysconfig import *; print "-I"+get_python_inc()')
 PY_LDLIBS = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_var as g; print " ".join(["-lpython"+g("VERSION"), g("SYSLIBS"), g("LIBS"), g("LOCALMODLIBS")])')
 
-CFLAGS = -fPIC -g -O2 -Wall -Wextra
+CFLAGS = -fPIC -fno-strict-aliasing -g -O2 -Wall -Wextra
 CPPFLAGS = $(PY_CPPFLAGS)
 LDFLAGS = $(PY_LDFLAGS)
 LDLIBS = $(PY_LDLIBS)
-- 
1.6.5

